kayak_core/src/context_ref.rs

```diff
@@ -260,6 +260,14 @@ impl<'a> KayakContextRef<'a> {
             .tree
             .merge(&tree, self.current_id.unwrap_or_default(), changes);
     }
+
+    /// This function is specifically for text rendering which needs to re-render when
+    /// it's parent layout is calculated.
+    pub fn mark_dirty(&mut self, widget_id: &Index) {
```

Comment:
Why can't we just get `widget_id` from `current_id`?

Reply:
Because I didn't think of it 👍